### DISCUSSION TITLE
Adventure map, Teach the AI to use adventure spells

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#7855] [Ideas]
Teach the AI to use adventure spells (View Mines, View Resources, View Artifacts, View Towns, View Heroes, View All) when the fog of war has not yet been removed to decide whether to explore a particular area.
For example, if the AI needs a gems mine for his level 6 creature (constructing the building, recruiting the creature), rather than discovering the adventure map a bit " at random ", the AI uses the view mines spell to head for the area where this mine is available.
For example, the AI could use the view heroes spell before planning the movement of its heroes, so that its weak heroes don't find themselves in the situation of getting too close to an opposing hero.
Etc.

Also, with the set elemental gardian spells, the AI could use them to protect its mines when weak opposing heroes are nearby (or not), forcing opponents to mobilize more resources if it wants to capture these mines.

The haunt spell could also be used by the AI to prevent the acquisition of mines by opponents, in cases where the AI has sufficient mines or resources of this type without reducing its development, or because it has no set elemental gardian spells, and the mine would be captured by an opponent anyway if it wasn't protected.

**Comment by oleg-derevenetz** [comment]
AI is already able to use the View All spell.

**Reply by ihhub** [reply]
This is in plans but as of now it is a very low priority. The amount of changes needed to implement the usage of these spells is too high for the very low outcome.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub,

What progress has been made on this subject?
So that I can update the original post or eventually close the discussion.
